pyproject.toml

```
@@ -74,7 +74,7 @@ exclude_lines = ["pragma: no cover", "if TYPE_CHECKING"]
 
 [tool.mypy]
 enable_error_code = "redundant-self"
-exclude = 'xarray/util/generate_.*\.py'
+exclude = 'xarray/util/generate_.*\.py|xarray/datatree_/.*\.py'
```

you could change `exclude` to become [a list of `str`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-exclude):
```suggestion
exclude = [
    'xarray/util/generate_.*\.py',
    'xarray/datatree_/.*\.py',
]
```

---

Thanks. I was going to try, (maybe did try?) multiple excludes, but this is much clearer.  I'll update that.